spec.md

```diff
@@ -629,6 +639,14 @@ message PluginCapability {
       // well as specific RPCs as indicated by
       // GroupControllerGetCapabilities.
       GROUP_CONTROLLER_SERVICE = 3 [(alpha_enum_value) = true];
+
+      // SNAPSHOT_METADATA_SERVICE indicates that the Plugin provides
+      // RPCs to retrieve metadata on the allocated blocks of a single
+      // snapshot, or the changed blocks between a pair of snapshots of
+      // the same block volume.
+      // The presence of this capability determines whether the CO will
+      // attempt to invoke the OPTIONAL SnapshotMetadata service RPCs.
+      SNAPSHOT_METADATA_SERVICE = 4 [(alpha_enum_value) = true];
```

Comment:
I blindly copied this pattern for the SnapshotMetadata service as it is an optional service.

Does this imply that if an SP wants to use a separate container for the SnapshotMetadata service that they'd have to also implement an Identity service in this container?

If this was indeed the case then the Kubernetes CO implementation (sidecar) could check for this.

Reply:
Feedback I received verbally on this is that every CSI server process must provide an Identity service.